storm-client/src/jvm/org/apache/storm/messaging/netty/NettyUncaughtExceptionHandler.java

```diff
@@ -33,10 +34,9 @@ public void uncaughtException(Thread t, Throwable e) {
 
         try {
             Utils.handleUncaughtException(e);
-        } catch (Throwable throwable) {
-            LOG.error("Exception thrown while handling uncaught exception " + throwable.getCause());
+        } catch (Error error) {
+            LOG.error("Exception thrown while handling uncaught exception " + error.getCause());
+            Runtime.getRuntime().exit(1);
         }
-        LOG.info("Received error in netty thread.. terminating server...");
```

Comment:
If a thread hits the exception handler will it still die? If so who is responsible for replacing it? As far as I can tell this handler is used by a HashedWheelTimer in Context, and it doesn't look to me like that class will replace a dead thread.

Reply:
It looks like the HashedWheelTimer will not be replaced - I'll take a second pass at this to confirm